src/fast_check/transform.rs

```diff
@@ -185,6 +186,7 @@ struct FastCheckTransformer<'a> {
   parsed_source: &'a ParsedSource,
   should_error_on_first_diagnostic: bool,
   diagnostics: Vec<FastCheckDiagnostic>,
+  public_inferred_namespaces: HashMap<String, Vec<ModuleItem>>,
```

Comment:
Right now this won't handle stuff like namespaces. Maybe this should instead be a stack of hashmaps where the top item represents the current module or namespace and then when we pop from the stack we add additional items to the namespace items.